internal/healthchecks/error.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthchecks
+
+// Error classification
+const (
+	Api        = "API"
+	Connection = "CONNECTION"
+	Generic    = "GENERIC"
+	Port       = "PORT"
+	Permission = "PERMISSION"
+)
+
+type HealthCheckError struct {
+	Code         string
+	Class        string
+	Message      string
+	Action       string
+	ResourceLink string
+	IsFatal      bool
+	Err          error
+}
+
+func (e HealthCheckError) Error() string {
+	return e.Err.Error()
+}
+
+var (
+	FbMetricsPortErr = HealthCheckError{
+		Code:         "FbMetricsPortErr",
+		Class:        Port,
+		Message:      "Port 20202 needed for fluent-bit self metrics is unavailable.",
+		Action:       "Verify that port 20202 is open.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	OtelMetricsPortErr = HealthCheckError{
+		Code:         "OtelMetricsPortErr",
+		Class:        Port,
+		Message:      "Port 20201 needed for OpenTelemetry self metrics is unavailable.",
+		Action:       "Verify that port 20201 is open.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	LogApiConnErr = HealthCheckError{
+		Code:         "LogApiConnErr",
+		Class:        Connection,
+		Message:      "Request to Logging API failed.",
+		Action:       "Check your internet connection and firewall rules.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	MonApiConnErr = HealthCheckError{
+		Code:         "MonApiConnErr",
+		Class:        Connection,
+		Message:      "Request to Monitoring API failed.",
+		Action:       "Check your internet connection and firewall rules.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
```

Comment:
Even though both the Logging and Monitoring pages ultimately have the same content, I think the URLs used here should be consistent with the error kind -- this is a Monitoring error so it should link to `https://cloud.google.com/monitoring/agent/ops-agent/troubleshooting`. Similarly, `LogApiScopeErr` and `LogApiPermissionErr` should be updated.

Reply:
Addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/055b441f8e3e7f5d6fa4d07388d10dc2c096497f .